Tighten unary op syntax and untupling language (#14372)
lampepfl/dotty: docs/docs/reference/changed-features/operators.md

This code is recognized as three different statements. `???` is syntactically a symbolic identifier, but
neither of its occurrences is followed by a space and a token that can start an expression.

## Unary operators

> As a minor restriction, the first parameter list of a unary operator must not be explicit and empty.

Reviewer: ```suggestion
A unary operator must not have explicit parameter lists even if they are empty.
```

A unary operator is a method named "unary_`op`" where `op` is one of `+`, `-`, `!`, or `~`.

> The reason for this restriction is that unary operator syntax does not support supplying an argument list, even if it is empty.

Reviewer: ```suggestion
```

Reviewer: I believe this one is redudant.
